### PR TITLE
Disk free space widget

### LINKED ISSUE #121: Disk free space widget
Hi! First of all, I love your project, using it on every mac I have. Appreciate your work!

**Is your feature request related to a problem? Please describe.**
The app I was using before iGlance is iStatMenus (if I'm not mistaken). There was nice widget (idk what term you use for that CPU, RAM, Network things on a bar). There were two letters F (fo Free space left on the disk) and U (for used place obviously). 
That thing was really useful for me. Would be happy to have such thing in iGlance as well.

**Describe the solution you'd like**
Want to see Free and Used disk space in GB

`**Describe alternatives you've considered**
iStatMenus

**Additional context**
Here is the screenshot from iStatMenus
image

And I can't stop thanking you for your brilliant app!

### PR DESCRIPTION
**Pull request by khorolets**
## Description

Adds widget about free and used disk space

image

## Related Issue

Closes #121

### CODE REVIEW

**Comment by khorolets** [comment]
@D0miH 

I'm asking to help with the build error

```
/Users/runner/runners/2.169.0/work/iGlance/iGlance/iGlance/iGlance/iGlance/MenuBarItems/MenuBarItemManager.swift:32:21: error: use of unresolved identifier 'DiskUsageMenuBarItem'
3373
    let diskUsage = DiskUsageMenuBarItem()
3374
                    ^~~~~~~~~~~~~~~~~~~~
```

Can't understand why it can't resolve identifier

**Comment by Moneypulation** [comment]
Hey,

thanks for your submission :) I've built your project and got the same error. It seems like this is caused because the swift classes you created aren't indexed in the xcodeproj file. Normally, when you create a class in your xcode project, it would add something like this (here with testclass.swift):

image

Did you create the class files outside of xcode, i.e. through the terminal? Because I couldn't find your classes indexed in your xcodeproj file. The problem should be fixed if you delete the class files from disk, create them through xcode (File -> New -> File.. -> Cocoa Class) and paste your code into them

**Comment by khorolets** [comment]
Hello @Moneypulation thanks. I believe it's because I haven't added that file to the commit, because I thought it contains changes I've made for "Signing..." settings to be able to run the project. I'll try to investigate it and update the PR.

**Comment by D0miH** [comment]
Great to hear that the issue is sorted out:)

I will look at the code in the afternoon today.

**Comment by khorolets** [comment]
@Moneypulation thank you once again, I've manually returned the team id in the file to keep things as it was before me. The build has passed :)

File: iGlance/iGlance/iGlance/MenuBarItems/DiskUsageMenuBarItem.swift [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    *          5_000 Bytes -> (value: "5", unit: "KB")
+    *          5_000_000 Bytes -> (value: "5", unit: "MB")
+    *          5_000_000_000 Bytes -> (value: "5", unit: "GB")
+    *
+    */
+    func convertToCorrectUnit(bytes: Int) -> (value: String, unit: String) {
+        if bytes < 1000 {
+            return (value: "\(bytes)", unit: "B")
+        }
+        let exp = Int(log2(Double(bytes)) / log2(1000.0))
```
Comment (D0miH) [inline iGlance/iGlance/iGlance/MenuBarItems/DiskUsageMenuBarItem.swift:67]: Wow, very elegant!

Comment (khorolets) [reply iGlance/iGlance/iGlance/MenuBarItems/DiskUsageMenuBarItem.swift:67]: Oh I’ve found this elegant solution on stackoverflow :)

**Comment by khorolets** [comment]
@D0miH thanks for the review, I'll make changes today or tomorrow

File: iGlance/iGlance/iGlance/MenuBarItems/DiskUsageMenuBarItem.swift [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    private func createMenuBarImage(up: (value: String, unit: String), down: (value: String, unit: String)) -> NSImage? {
+        // create the attributed strings for the upload and download
+        let freeString = self.crateAttributedString(text: "F:")
+        let usedString = self.crateAttributedString(text: "U:")
+        let freeSpaceString = self.createAttributedBandwidthString(value: down.value, unit: down.unit)
+        let usedSpaceString = self.createAttributedBandwidthString(value: up.value, unit: up.unit)
+
+        // create the menu bar image for the bandwidth.
+        let textWidth = CGFloat(55) // this value was found by trail and error
+        let iconWidth = CGFloat(12)
```
[the remark this answers is not in the document]
Comment (khorolets) [reply iGlance/iGlance/iGlance/MenuBarItems/DiskUsageMenuBarItem.swift]: I hope I got your proposal right

**Comment by D0miH** [comment]
Thanks for your contribution @khorolets! 👍🏼☺️
Feel free to help us again in the future :)

**Comment by khorolets** [comment]
@D0miH it was a pleasure. I'll try to help in future if I am able to :) Thank you for your work!

**Pull request merged**